egs/librispeech/asr/simple_v1/decode.py

```diff
@@ -47,22 +51,36 @@ def decode(dataloader: torch.utils.data.DataLoader, model: AcousticModel,
         nnet_output = nnet_output.permute(0, 2,
                                           1)  # now nnet_output is [N, T, C]
 
+        print('about convert to dense')
+        print(LG.shape)
+        print(LG.labels.shape)
         dense_fsa_vec = k2.DenseFsaVec(nnet_output, supervision_segments)
-        assert LG.is_cuda()
+        # assert LG.is_cuda()
         assert LG.device == nnet_output.device, \
             f"Check failed: LG.device ({LG.device}) == nnet_output.device ({nnet_output.device})"
         # TODO(haowen): with a small `beam`, we may get empty `target_graph`,
         # thus `tot_scores` will be `inf`. Definitely we need to handle this later.
-        lattices = k2.intersect_dense_pruned(LG, dense_fsa_vec, 2000.0, 20.0,
-                                             30, 300)
+        print('about to intersect')
+        lattices = k2.intersect_dense_pruned(LG, dense_fsa_vec, 20.0, 20.0, 30,
+                                             300000)
+        print(LG.shape)
+        print(dense_fsa_vec.dim0())
+        # lattices = k2.intersect_dense(LG, dense_fsa_vec, 10.0)
+        print('about to get shortest path')
         best_paths = k2.shortest_path(lattices, use_float_scores=True)
+        print('about to get shortest path to cpu')
         best_paths = best_paths.to('cpu')
         assert best_paths.shape[0] == len(texts)
 
+        print('about to output')
         for i in range(len(texts)):
-            hyp_words = [
-                symbols.get(x) for x in best_paths[i].aux_labels if x > 0
-            ]
+            if isinstance(best_paths[i].aux_labels, torch.Tensor):
+                aux_labels = best_paths[i].aux_labels
+            else:
+                # it's a ragged tensor
```

Comment:
In which case will it be a ragged tensor?

Reply:
e.g. in `determinize`

Reply:
would add the comments in the next PR